bnd-indexer-maven-plugin:7.0.0-SNAPSHOT:index goal fails due to "JAR/ZIP file ... seems corrupted"

I tried a recent 7.0.0-SNAPSHOT on openhab-core but it fails when running the `index` goal:

`[ERROR] Failed to execute goal biz.aQute.bnd:bnd-indexer-maven-plugin:7.0.0-SNAPSHOT:index (index) on project org.openhab.core.bom.runtime-index: illegal format /home/wouter/git/openhab/openhab-core/bom/runtime/pom.xml: The JAR/ZIP file (/home/wouter/git/openhab/openhab-core/bom/runtime/pom.xml) seems corrupted, error: zip END header not found -> [Help 1]`

These are the POMs:

* https://github.com/wborn/openhab-core/blob/bnd-7/bom/runtime/pom.xml
* https://github.com/wborn/openhab-core/blob/bnd-7/bom/runtime-index/pom.xml

See also this GitHub Actions build: https://github.com/wborn/openhab-core/actions/runs/5712301962

This is the full stack trace when I build using `-X`:

```
[ERROR] Failed to execute goal biz.aQute.bnd:bnd-indexer-maven-plugin:7.0.0-SNAPSHOT:index (index) on project org.openhab.core.bom.runtime-index: illegal format /home/wouter/git/openhab/openhab-core/bom/runtime/pom.xml: The JAR/ZIP file (/home/wouter/git/openhab/openhab-core/bom/runtime/pom.xml) seems corrupted, error: zip END header not found -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal biz.aQute.bnd:bnd-indexer-maven-plugin:7.0.0-SNAPSHOT:index (index) on project org.openhab.core.bom.runtime-index: illegal format /home/wouter/git/openhab/openhab-core/bom/runtime/pom.xml
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:333)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.builder.multithreaded.MultiThreadedBuilder$1.call (MultiThreadedBuilder.java:193)
    at org.apache.maven.lifecycle.internal.builder.multithreaded.MultiThreadedBuilder$1.call (MultiThreadedBuilder.java:180)
    at java.util.concurrent.FutureTask.run (FutureTask.java:264)
    at java.util.concurrent.Executors$RunnableAdapter.call (Executors.java:539)
    at java.util.concurrent.FutureTask.run (FutureTask.java:264)
    at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1136)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:635)
    at java.lang.Thread.run (Thread.java:833)
Caused by: org.apache.maven.plugin.MojoExecutionException: illegal format /home/wouter/git/openhab/openhab-core/bom/runtime/pom.xml
    at aQute.bnd.maven.indexer.plugin.IndexerMojo.execute (IndexerMojo.java:211)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:126)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:328)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.builder.multithreaded.MultiThreadedBuilder$1.call (MultiThreadedBuilder.java:193)
    at org.apache.maven.lifecycle.internal.builder.multithreaded.MultiThreadedBuilder$1.call (MultiThreadedBuilder.java:180)
    at java.util.concurrent.FutureTask.run (FutureTask.java:264)
    at java.util.concurrent.Executors$RunnableAdapter.call (Executors.java:539)
    at java.util.concurrent.FutureTask.run (FutureTask.java:264)
    at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1136)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:635)
    at java.lang.Thread.run (Thread.java:833)
Caused by: java.lang.IllegalArgumentException: illegal format /home/wouter/git/openhab/openhab-core/bom/runtime/pom.xml
    at aQute.bnd.osgi.resource.ResourceBuilder.parse (ResourceBuilder.java:1039)
    at aQute.bnd.osgi.resource.ResourceBuilder.lambda$addFile$8 (ResourceBuilder.java:995)
    at aQute.bnd.osgi.resource.FileResourceCache.lambda$getResource$1 (FileResourceCache.java:65)
    at java.util.concurrent.ConcurrentHashMap.computeIfAbsent (ConcurrentHashMap.java:1740)
    at aQute.bnd.osgi.resource.FileResourceCache.getResource (FileResourceCache.java:65)
    at aQute.bnd.osgi.resource.ResourceBuilder.addFile (ResourceBuilder.java:995)
    at aQute.bnd.maven.indexer.plugin.IndexerMojo.execute (IndexerMojo.java:182)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:126)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:328)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.builder.multithreaded.MultiThreadedBuilder$1.call (MultiThreadedBuilder.java:193)
    at org.apache.maven.lifecycle.internal.builder.multithreaded.MultiThreadedBuilder$1.call (MultiThreadedBuilder.java:180)
    at java.util.concurrent.FutureTask.run (FutureTask.java:264)
    at java.util.concurrent.Executors$RunnableAdapter.call (Executors.java:539)
    at java.util.concurrent.FutureTask.run (FutureTask.java:264)
    at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1136)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:635)
    at java.lang.Thread.run (Thread.java:833)
Caused by: java.util.zip.ZipException: The JAR/ZIP file (/home/wouter/git/openhab/openhab-core/bom/runtime/pom.xml) seems corrupted, error: zip END header not found
    at aQute.bnd.osgi.Jar.buildFromZip (Jar.java:311)
    at aQute.bnd.osgi.Jar.<init> (Jar.java:137)
    at aQute.bnd.osgi.Jar.<init> (Jar.java:235)
    at aQute.bnd.osgi.resource.ResourceBuilder.parse (ResourceBuilder.java:1024)
    at aQute.bnd.osgi.resource.ResourceBuilder.lambda$addFile$8 (ResourceBuilder.java:995)
    at aQute.bnd.osgi.resource.FileResourceCache.lambda$getResource$1 (FileResourceCache.java:65)
    at java.util.concurrent.ConcurrentHashMap.computeIfAbsent (ConcurrentHashMap.java:1740)
    at aQute.bnd.osgi.resource.FileResourceCache.getResource (FileResourceCache.java:65)
    at aQute.bnd.osgi.resource.ResourceBuilder.addFile (ResourceBuilder.java:995)
    at aQute.bnd.maven.indexer.plugin.IndexerMojo.execute (IndexerMojo.java:182)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:126)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:328)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.builder.multithreaded.MultiThreadedBuilder$1.call (MultiThreadedBuilder.java:193)
    at org.apache.maven.lifecycle.internal.builder.multithreaded.MultiThreadedBuilder$1.call (MultiThreadedBuilder.java:180)
    at java.util.concurrent.FutureTask.run (FutureTask.java:264)
    at java.util.concurrent.Executors$RunnableAdapter.call (Executors.java:539)
    at java.util.concurrent.FutureTask.run (FutureTask.java:264)
    at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1136)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:635)
    at java.lang.Thread.run (Thread.java:833)
Caused by: java.util.zip.ZipException: zip END header not found
    at java.util.zip.ZipFile$Source.findEND (ZipFile.java:1599)
    at java.util.zip.ZipFile$Source.initCEN (ZipFile.java:1607)
    at java.util.zip.ZipFile$Source.<init> (ZipFile.java:1445)
    at java.util.zip.ZipFile$Source.get (ZipFile.java:1407)
    at java.util.zip.ZipFile$CleanableResource.<init> (ZipFile.java:716)
    at java.util.zip.ZipFile.<init> (ZipFile.java:250)
    at java.util.zip.ZipFile.<init> (ZipFile.java:179)
    at java.util.zip.ZipFile.<init> (ZipFile.java:193)
    at aQute.bnd.osgi.Jar.buildFromZip (Jar.java:300)
    at aQute.bnd.osgi.Jar.<init> (Jar.java:137)
    at aQute.bnd.osgi.Jar.<init> (Jar.java:235)
    at aQute.bnd.osgi.resource.ResourceBuilder.parse (ResourceBuilder.java:1024)
    at aQute.bnd.osgi.resource.ResourceBuilder.lambda$addFile$8 (ResourceBuilder.java:995)
    at aQute.bnd.osgi.resource.FileResourceCache.lambda$getResource$1 (FileResourceCache.java:65)
    at java.util.concurrent.ConcurrentHashMap.computeIfAbsent (ConcurrentHashMap.java:1740)
    at aQute.bnd.osgi.resource.FileResourceCache.getResource (FileResourceCache.java:65)
    at aQute.bnd.osgi.resource.ResourceBuilder.addFile (ResourceBuilder.java:995)
    at aQute.bnd.maven.indexer.plugin.IndexerMojo.execute (IndexerMojo.java:182)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:126)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:328)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.builder.multithreaded.MultiThreadedBuilder$1.call (MultiThreadedBuilder.java:193)
    at org.apache.maven.lifecycle.internal.builder.multithreaded.MultiThreadedBuilder$1.call (MultiThreadedBuilder.java:180)
    at java.util.concurrent.FutureTask.run (FutureTask.java:264)
    at java.util.concurrent.Executors$RunnableAdapter.call (Executors.java:539)
    at java.util.concurrent.FutureTask.run (FutureTask.java:264)
    at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1136)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:635)
    at java.lang.Thread.run (Thread.java:833)
```

--- a/maven-plugins/bnd-indexer-maven-plugin/src/main/java/aQute/bnd/maven/indexer/plugin/IndexerMojo.java
+++ b/maven-plugins/bnd-indexer-maven-plugin/src/main/java/aQute/bnd/maven/indexer/plugin/IndexerMojo.java
@@ -178,6 +178,12 @@ public class IndexerMojo extends AbstractMojo {
 			IO.mkdirs(outputFile.getParentFile());
 			for (Entry<File, ArtifactResult> entry : dependencies.entrySet()) {
 				File file = entry.getKey();
+
+				if (!"jar".equals(entry.getValue().getArtifact().getExtension())) {
+					logger.debug("Skipping: {}", file);
+					continue;
+				}
+
 				ResourceBuilder resourceBuilder = new ResourceBuilder();
 				resourceBuilder.addFile(entry.getKey(), repositoryURLResolver.resolver(file, entry.getValue()));
 

